File: Events.cs
public class Counter
{
    public event EventHandler FirstEvent;

    public void RaiseFirstEvent()
    {
        EventHandler handler = FirstEvent;

        handler?.Invoke(this, EventArgs.empty());
    }
}

public class Program
{
    static void Main()
    {
        var counter = new Counter();

        counter.FirstEvent += (object sender, EventArgs e) => {
			Console.WriteLine("Event!.");
        };
    }
}

File: Events.java
interface Listener extends EventListener {
    void notify(String message);
}

public class Counter {
	private Listener listener;

	public void addListener(Listener listener) {
		this.listener = listener;
	}

	public void RaiseFirstEvent() {
		if(this.listener != null) {
			this.listener.notify("Event!");
		}
	} 
}

public class Program {
    static void Main()
    {
        var counter = new Counter();

        counter.addListener(new Listener() {
    		@Override
            public void notify(String message) {
                System.out.println(message);
            }
        });
    }
}